### PR TITLE
Merge dev15-preview-5 into master

### PR DESCRIPTION
**Pull request by vslsnap**
This is an automatically generated pull request from dev15-preview-5 into master.

@dotnet/roslyn-infrastructure:

``` bash
git remote add vslsnap-roslyn "https://github.com/vslsnap/roslyn.git"
git fetch vslsnap-roslyn
git fetch upstream
git checkout merge-dev15-preview-5-into-master20160923-150132
git reset --hard upstream/master
git merge upstream/dev15-preview-5
# Fix merge conflicts
git commit
git push vslsnap-roslyn merge-dev15-preview-5-into-master20160923-150132 --force
```

Once the merge can be made and all the tests pass, you are free to merge the pull request.

### CODE REVIEW

**Comment by vslsnap** [comment]
@dotnet-bot test vsi please

**Comment by jasonmalinowski** [comment]
@Pilchie and @heejaechang I had to do a bit of manual merge resolution here to get the deferred project load stuff to merge with the moving of VisualStudio.Next. If either of you want to take a look that might be good just to make sure nothing looks terribly bad.

**Comment by Pilchie** [comment]
My parts look right to me.

**Comment by jasonmalinowski** [comment]
@dotnet retest windows_debug_unit64_prtest please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_eta_open_prtest please. This was broken earlier today due to a bad cross-cutting change.

File: src/VisualStudio/Core/Next/Remote/RemoteHostClientFactory.cs [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Remote
 {
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
```
Comment (heejaechang) [inline src/VisualStudio/Core/Next/Remote/RemoteHostClientFactory.cs]: what is this for?

Comment (heejaechang) [inline src/VisualStudio/Core/Next/Remote/RemoteHostClientFactory.cs]: ah,... there is now 2 workspace?

**Comment by heejaechang** [comment]
other .Next change looks fine.

File: src/VisualStudio/Core/Next/Remote/RemoteHostClientFactory.cs (hunk 1 above)
Comment (Pilchie) [inline src/VisualStudio/Core/Next/Remote/RemoteHostClientFactory.cs]: Yes. I will try to convert the other reference to an extern alias to avoid this.

**Pull request merged**